### ISSUE TITLE
[P1] NeuralMie optics returns NaN mass gradients in BC-free modes

### ISSUE THREAD

**Issue by aliakherati** [climate-analytics-lab/jax-gcm#793]
Found reviewing #792 / #791 at PR head `65e8e78201dae785c35a67a533f829b9d0338b2a`. Reproduced on CPU with JAX 0.11.1.

`jcm/physics/aerosol/jam/optics/optics_term.py:322` evaluates:

```python
f_core = jnp.cbrt(jnp.clip(v_core / safe_tot, 0.0, 1.0))
```

For a mode with zero BC, the cube-root derivative is singular at zero. The mode still takes the core-shell network because selection tests static species membership rather than actual BC presence. Reverse mode produces NaNs even for derivatives with respect to another species' positive mass. The later dry-mass gate cannot repair unsafe derivatives already evaluated upstream.

Reproducer from the JAX-GCM checkout, using the existing fixture and recomputing physically consistent modal geometry:

```python
import jax
import jax.numpy as jnp
from jcm.physics.aerosol.jam import MAM4_SPEC, mass_name
from jcm.physics.aerosol.jam.microphysics.placeholder import equilibrium_modal_state
from jcm.physics.aerosol.jam.optics.optics_term import JamOpticsTerm
from jcm.physics.aerosol.jam.optics.optics_term_test import _setup

state, diagnostics, band, *_ = _setup(nlev=1, ncols=1, n_sw=1, n_lw=0)
term = JamOpticsTerm(optics_backend='neuralmie')
term.cache_band_config(band)
tracers = dict(state.tracers)
for mode in MAM4_SPEC.modes:
    if 'bc' in mode.species:
        tracers[mass_name('bc', mode.short)] = jnp.zeros((1, 1))
key = mass_name('so4', 'acc')

def loss(scale):
    ts = dict(tracers)
    ts[key] = ts[key] * scale
    ds = dict(diagnostics)
    ds['_jam_state'] = equilibrium_modal_state(
        ts, ts, MAM4_SPEC, jnp.full((1, 1), .5))
    _, diag = term(state.copy(tracers=ts), ds, None, None)
    return jnp.sum(diag['aerosol'].aod_sw_per_band)

print(loss(1.0))  # finite positive AOD
print(jax.grad(loss)(1.0))  # nan
print((loss(1.001) - loss(.999)) / .002)  # finite sensitivity
```

Guard the input before the cube root, with an explicit zero-core derivative policy and a valid no-core limit. Merely wrapping the resulting cube root or network output in `where` does not make its derivative safe. Add end-to-end gradients for BC-free populated modes, completely empty modes, and near-zero positive BC, covering extinction, absorption, and diagnostics. The existing mass-gradient test populates every species and misses this boundary.
